docs/DeveloperGuide.md

```diff
@@ -1,34 +1,851 @@
+1. The `addTask` method calls the `checkIfModuleValid` method in `ModuleList` class to check if `Section 2` is a 
+valid module code.
+    * If module code is not valid, the `addTask` method will return and `Ui` will prompt the user to enter a 
+    valid module code.
+    * Else if module code is valid, the `addTask` method will execute the next step.
+1. The `addTask` method calls the `checkIfModuleExist` method to check if `Secion 2` (module code) exists in the database.
+    * If module code does not exist in the database, the `addTask` method will return and `Ui` will prompt the user to
+    execute the `addmod` or `addexp` command first.
+    * Else if module code exists, the `addTask` method will execute the next step.
+1. The `addTask` method will create a new `Task` object by the constructor in `Task` class, with `Section 3` (task description). 
+1. The `addTask` method adds the `Task` object to the array list `tasks`.
+
+![addtask](diagrams/addtask.png)
+
+#### Future Implementation
+A future implementation requires user to enter the expected time required to complete the task. The `addTask` method will split the 
+user input into 4 sections, with `Section 4` as the expected time required to complete the task. `Section 4` will then be
+used in the future implementation of mark task as done feature, as further illustrated in the [mark task as done](#mark-task-as-done)
+section.
+
+### 4.8 Mark Task as Done <a name="4.8"></a>
+The mark task as done feature allows user to mark an existing task as done.
+
+#### Current Implementation
+The `done` command is executed by the `parse` method in the `Parser` class, which then calls the `setDone` method in the
+`TaskList` class. The `setDone` method performs all the necessary logic for the mark task as done feature.
+
+Given below is an example usage scenario.
+1. The user input the `done` command and the `parse` method in `Parser` parses the command.
+1. `parse` calls the `setDone` method in `TaskList`.
+1. The `setDone` method splits the user input into 2 sections:
+    * `Section 1`: command
+    * `Section 2`: task number corresponding to the index of `Task` object in array list `tasks`
+1. The `setDone` method will check if the array list `tasks` is empty.
+    * If array list is empty, `Ui` will notify the user that there is no `Task` to mark as done.
+    * Else if array list is not empty, `setDone` method will execute the next step.
+1. The `setDone` method will check the validity of `Section 2` (task number).
+    * If task number is not an integer, an exception will be thrown and the `setDone` method terminates.
+    * If task number is out of bounds, `Ui` will prompt the user to enter a valid task number.
+    * Else if task number is valid, `setDone` method will execute the next step.
+1. The `setDone` method will then call the `setAsDone` method in `Task` class to mark the `Task` corresponding to 
+`Section 2` (task number) as done. 
+    
+![markdone](diagrams/markdone.png)  
```

Comment:
Are you missing a return arrow here?
![review 8](https://user-images.githubusercontent.com/57140494/97387504-17d67080-1911-11eb-8a96-aa04af3633f2.png)